lib/pscom4ivshmem/direct_mode/psivshmem_malloc.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * ParaStation
+ *
+ * Copyright (C) 2013 ParTec Cluster Competence Center GmbH, Munich
+ *
+ * All rights reserved.
+ *
+ * Author:	Jens Hauke <hauke@par-tec.com>
+ */
+
+#ifndef _PSIVSHMEM_MALLOC_H_
+#define _PSIVSHMEM_MALLOC_H_
+
+
+#include <stddef.h>
+
+struct Psivshmem {
+	void *base; /* base pointer of shared mem segment */
+	long baseoffset;
+	void *end;  /* = base + size */
+	void *tail;
+	size_t size;
+//	int ivshmemid; /* ivshmemid of shared mem segment at base */
+	const char *msg; /* Message if initialization failed */
+};
+
+/* Get the Psivshmem of the shared memory. */
+extern struct Psivshmem psivshmem_direct_info;
+
+
+/* Check if the Pointer ptr is part of the shared memory */
+static inline
+int is_psivshmem_ptr(void *ptr)
```

Comment:
What is the purpose of this function? Is is still to be implemented? In that case, place a TODO with an according description.